compiler/il/OMRNode.cpp

```diff
@@ -4492,6 +4492,8 @@ OMR::Node::getEvaluationPriority(TR::CodeGenerator * cg)
 int32_t
 OMR::Node::setEvaluationPriority(int32_t p)
    {
+   TR_ASSERT(!self()->getOpCode().isTreeTop(), "cannot set evaluation priority of a treetop");
```

Comment:
Similarly, may I ask you to add a comment indicating that setting the evaluation priority is only meaningful for nodes that yields values, and tree-top nodes do not yield values?  And then further note that `_unionA` is used for a different purpose for guards nodes, and using it evaluation priority would conflict with that use.